.github/workflows/sonarcloud.yml

```
@@ -15,56 +15,53 @@ jobs:
-      - name: Detect Sonar token
-        id: sonar-token
-        shell: bash
-        run: |
-          if [ -n "${{ secrets.SONAR_TOKEN }}" ]; then
-            echo "enabled=true" >> "$GITHUB_OUTPUT"
-          else
-            echo "enabled=false" >> "$GITHUB_OUTPUT"
-            echo "SONAR_TOKEN is not configured; skipping SonarCloud scan."
-          fi
+      - name: Sonar token not configured
+        if: ${{ secrets.SONAR_TOKEN == '' }}
```

actionlint (1.7.11)

[error] context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)